src/Auth.ts

```diff
@@ -3,10 +3,12 @@ import type * as NodeForge from 'node-forge';
 import { FileTokenStorage } from './auth/FileTokenStorage';
 import { TokenStorage } from './auth/TokenStorage';
 import type { AuthServer } from './AuthServer';
-import { Logger } from './cli';
+import { Cli, Logger } from './cli';
 import { CommandError } from './Command';
 import config from './config';
 import request from './request';
+import { settingsNames } from './settingsNames';
+import * as open from 'open';
```

Comment:
Ideally, we should lazy-load `open` so that we load it only when needed rather than on each CLI execution. If you find it too complicated, I can do it when merging the PR.

Reply:
> Hey @michaelmaillot, we're about to replace ADAL with MSAL (#2303) which will likely affect this PR. Would you mind waiting until we merged it and do the changes then? It would help us avoid doing the changes twice. Thank you! ❤️

No pb, I'll wait for your go to push my corrections!

Reply:
Ok looks like it's not as easy as I though.

The error met in Windows test is regarding mocking the `open` call, which I can't stub. Which is why I got a spawned (child process) error when calling `getDeviceCodeResponse`.

The fact is that if I dynamically load the `open` module, I can't stub it, as it's a standalone one. And as I don't put it in a private property from which I could mock the call with a stub (like it's done in `AuthServer.ts`), I can't figure out how to mock it.